pkg/kosmosctl/rsmigrate/common.go

```diff
@@ -0,0 +1,60 @@
+package rsmigrate
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/kosmos.io/kosmos/pkg/apis/kosmos/v1alpha1"
+	kosmosversioned "github.com/kosmos.io/kosmos/pkg/generated/clientset/versioned"
+)
+
+func getClientFromKnode(knode *v1alpha1.Knode) (kubernetes.Interface, *kosmosversioned.Clientset, error) {
+	//generate clientset by knode kubeconfig
+	knodeKubeconfig := knode.Spec.Kubeconfig
+	if len(knodeKubeconfig) == 0 {
+		return nil, nil, fmt.Errorf("knode's kubeconfig is nil, it's unable to work normally")
+	}
+
+	knodeConfig, err := clientcmd.NewClientConfigFromBytes(knodeKubeconfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("get knode client config error: %s", err)
+	}
+	knodeRestconfig, err := knodeConfig.ClientConfig()
+	if err != nil {
+		return nil, nil, fmt.Errorf("get knode rest client config error:%s", err)
+	}
+	k8sClient, err := kubernetes.NewForConfig(knodeRestconfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("create kubernetes clientset error: %s ", err)
+	}
+	crdClient, err := kosmosversioned.NewForConfig(knodeRestconfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("create kubernetes clientset for knode crd error: %s ", err)
+	}
+
+	return k8sClient, crdClient, nil
+}
```

Comment:
use use utils.NewKosmosClientFromBytes to cleanup the code

Reply:
modified